src/arch/linux/mapping/mapping.c

```diff
@@ -464,6 +464,15 @@ void *mmap_mapping(int cap, dosaddr_t targ, size_t mapsize, int protect)
 
   Q__printf("MAPPING: map, cap=%s, target=%p, size=%zx, protect=%x\n",
 	cap, target, mapsize, protect);
+#ifdef __LP64__
+  if (targ == (dosaddr_t)-1 && _MAP_32BIT == 0) {
+    /* make sure it's allocated within DOS addressable range */
+    target = mapping_find_hole((uintptr_t)MEM_BASE32(0),
```

Comment:
You probably need to force
MAP_FIXED_NOREPLACE here to
not race with some thread?